javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java

```diff
@@ -1164,11 +1164,15 @@ private void printSwitchNode(SwitchNode n, Void arg) {
         n.getSelector().accept(this, arg);
         printer.println(") {");
         if (n.getEntries() != null) {
-            printer.indent();
+            if(configuration.isIndentInSwitch()) {
+                printer.indent();
+            }
             for (final SwitchEntry e : n.getEntries()) {
                 e.accept(this, arg);
             }
-            printer.unindent();
+            if(configuration.isIndentInSwitch()) {
+                printer.unindent();
+            }
         }
```

Comment:
Or maybe it is better to write it as 
```java 
            if(configuration.isIndentInSwitch()) {
                printer.indent();
                for (final SwitchEntry e : n.getEntries())
                    e.accept(this, arg);
                print.unindent();
            }else{
                for (final SwitchEntry e : n.getEntries()) 
                    e.accept(this, arg);
            }
```

Reply:
Personally, I think a cleaner/tidier alternative would be:

```java
printer.indent(configuration.isIndentInSwitch());
```

Or a helper method `indentIf(boolean shouldIndent)` (or similar).

Reply:
Good idea I will do that

Reply:
I don't see why the indenting printer should be modified for a single case?